modules/core/src/main/java/org/apache/synapse/FaultHandler.java

```diff
@@ -135,4 +143,23 @@ private void traceOrDebugWarn(boolean traceOn, String msg) {
         }
         log.warn(msg);
     }
+
+    /**
+     * Execute the ExtendedSynapseHandler in the error flow.
+     *
+     * @param synCtx Synapse Message Context
+     */
+    protected void executeExtendedSynapseHandler(MessageContext synCtx) {
+        List handlers = synCtx.getEnvironment().getSynapseHandlers();
+        Iterator<SynapseHandler> iterator = handlers.iterator();
+        while (iterator.hasNext()) {
+            SynapseHandler handler = iterator.next();
+            if ((handler instanceof AbstractExtendedSynapseHandler) &&
+                    (synCtx.getProperty(SynapseConstants.IS_ERROR_COUNT_ALREADY_PROCESSED) == null)) {
+                if (!((AbstractExtendedSynapseHandler) handler).handleError(synCtx)) {
```

Comment:
Shall we remove these unnecessary brackets, code will be bit hard to read.  Please do the same in other places as well.

Reply:
Fixed